Tests/MongoSwiftTests/MongoCollectionTests.swift

```diff
@@ -307,17 +307,31 @@ final class MongoCollectionTests: MongoSwiftTestCase {
         expect(try coll1.findOneAndDelete(["x": 4])).to(equal(b4))
     }
 
-    func testEncodeCursorType() throws {
+    func testCursorType() throws {
         let encoder = BSONEncoder()
 
-        let nonTailable = FindOptions(cursorType: .nonTailable)
+        var nonTailable = FindOptions(cursorType: .nonTailable)
         expect(try encoder.encode(nonTailable)).to(equal(["awaitData": false, "tailable": false]))
 
-        let tailable = FindOptions(cursorType: .tailable)
+        // test mutated cursorType
+        nonTailable.cursorType = .tailable
+        expect(try encoder.encode(nonTailable)).to(equal(["awaitData": false, "tailable": true ]))
+
+        var tailable = FindOptions(cursorType: .tailable)
         expect(try encoder.encode(tailable)).to(equal(["awaitData": false, "tailable": true ]))
 
-        let tailableAwait = FindOptions(cursorType: .tailableAwait)
+        tailable.cursorType = .nonTailable
+        expect(try encoder.encode(tailable)).to(equal(["awaitData": false, "tailable": false ]))
+
+        var tailableAwait = FindOptions(cursorType: .tailableAwait)
         expect(try encoder.encode(tailableAwait)).to(equal(["awaitData": true, "tailable": true ]))
+
+        tailableAwait.cursorType = .tailable
+        expect(try encoder.encode(tailableAwait)).to(equal(["awaitData": false, "tailable": true ]))
+
+        // test nill cursorType
+        var nilTailable = FindOptions(cursorType: nil)
```

Comment:
since this doesn't get modified, you can make it a `let` constant. (the compiler will generate a warning about this not needing to be mutable every time we build the tests.)

however, it might be worth keeping it mutable, and adding a few more lines to this test to also check what happens when `cursorType` is initially `nil` and is later set to something. (and maybe vice versa, where the user initially provided a cursorType and then the user sets it to nil.)